src/rpc/rewards.cpp

```diff
@@ -0,0 +1,883 @@
+                    }
+                }
+                pendingTxn.payments.clear();
+            }
+
+            if (errorsOccurred) {
+                responseObj.push_back(Pair("error_rejects", "One or more transactions were rejected"));
+            }
+
+            responseObj.push_back(Pair("batch_results", batchResults));
+
+            //  Write the payment back to the database if anything succeeded
+            if (atLeastOneTxnSucceeded) {
+                if (!pPayoutDb->UpdatePayoutEntry(payoutEntry)) {
+                    LogPrintf("Failed to update payout DB payment status for reward '%s'!\n",
+                        payoutEntry.rewardID.c_str());
+
+                    responseObj.push_back(Pair("error_db_update", "Payout DB update failed"));
+                }
+            }
+            else {
+                LogPrintf("All transactions failed for reward '%s'!\n",
+                    payoutEntry.rewardID.c_str());
+            }
+        }
+
+        return responseObj;
+    }
+
+    throw JSONRPCError(RPC_DATABASE_ERROR, std::string("Failed to execute payments specified reward"));
+}
+
+static const CRPCCommand commands[] =
+    {           //  category    name                          actor (function)             argNames
+                //  ----------- ------------------------      -----------------------      ----------
+            {   "rewards",      "schedulereward",             &schedulereward,             {"total_payout_amount", "payout_source", "target_asset_name", "exception_addresses"}},
+            {   "rewards",      "getreward",                  &getreward,                  {"reward_id"}},
+            {   "rewards",      "cancelreward",               &cancelreward,               {"reward_id"}},
+            {   "rewards",      "calculatepayments",          &calculatepayments,          {"reward_id"}},
+            {   "rewards",      "getpayments",                &getpayments,                {"reward_id"}},
+            {   "rewards",      "cancelpayments",             &cancelpayments,             {"reward_id"}},
+            {   "rewards",      "executepayments",            &executepayments,            {"reward_id"}},
+    };
+
+void RegisterRewardsRPCCommands(CRPCTable &t)
+{
+    for (unsigned int vcidx = 0; vcidx < ARRAYLEN(commands); vcidx++)
+        t.appendCommand(commands[vcidx].name, &commands[vcidx]);
+}
+
+CAmount AmountFromValue(bool p_isRVN, const UniValue& p_value)
+{
+    if (!p_value.isNum() && !p_value.isStr())
+        throw JSONRPCError(RPC_TYPE_ERROR, "Amount is not a number or string");
+
+    CAmount amount;
+    if (!ParseFixedPoint(p_value.getValStr(), 8, &amount))
+        throw JSONRPCError(RPC_TYPE_ERROR, strprintf("Invalid amount (3): %s", p_value.getValStr()));
+
+    if (p_isRVN && !MoneyRange(amount))
+        throw JSONRPCError(RPC_TYPE_ERROR, strprintf("Amount out of range: %s", amount));
+
+    return amount;
+}
+
```

Comment:
I think this function already exists. However, it doesn't take into the account of the p_isRVN bool. Maybe we should it, so we aren't duplicating code.

Reply:
Done